### PR TITLE
Add GitFiles command

### PR DESCRIPTION
**Pull request by icholy**

### CODE REVIEW

**Comment by junegunn** [comment]
Thanks. We can already take advantage of `git ls-tree` using `$FZF_DEFAULT_COMMAND` (https://github.com/junegunn/fzf#git-ls-tree-for-fast-traversal), but yeah, I think it's okay to have a dedicated command for it. How about changing the prompt to `GitFiles>`?

**Comment by junegunn** [comment]
And please update doc/fzf-vim.txt file as well.

File: autoload/fzf/vim.vim [hunk 1]
```diff
@@ -324,6 +324,18 @@ function! fzf#vim#history(...)
 endfunction
 
 " ------------------------------------------------------------------
+" GitFiles
+" ------------------------------------------------------------------
+
+function! fzf#vim#gitfiles(...)
+  call s:fzf({
+  \ 'source':  'git ls-tree --name-only --full-tree -r HEAD',
```
Comment (junegunn) [inline autoload/fzf/vim.vim]: I don't think we should use `--full-tree`. Vim will unable to find the file if the current working directory is not the root of the repo.

**Comment by icholy** [comment]
Do you think it should it also a take a `[path]` parameter like the `Files` command?

**Comment by junegunn** [comment]
I don't think it's necessary. But I think we should check if we're currently under a git repo and print warning message if not instead of showing the empty result. (like `:Snippets` command when ultisnips is unavailable)

**Comment by junegunn** [comment]
Merged, thanks. I also added the warning message and applied `dir` option which makes fzf start from the root of the repository so that we can see any file on the repository and open it regardless of the current working directory.

**Pull request merged**